jetty-util/src/main/java/org/eclipse/jetty/util/resource/FileResource.java

```diff
@@ -204,13 +212,21 @@ public Resource addPath(String path)
         }
         catch(final URISyntaxException e)
         {
-            throw new MalformedURLException(){{initCause(e);}};
+            throw new InvalidPathException(path, e.getMessage());
         }
 
         return new FileResource(uri);
     }
-   
-    
+
+    private void assertValidPath(String path)
+    {
+        Matcher mat = CNTRL_PATTERN.matcher(path);
```

Comment:
This is an expensive test to have on every Resource creation.  We really need to avoid creating RegEx matchers on common paths.